spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderUtils.java

```diff
@@ -59,33 +70,73 @@ static void prepareConsumerBinding(String name, String group,
 		}
 
 		if (extendedConsumerProperties.getExtension().isEnableDlq()) {
-			KafkaStreamsDlqDispatch kafkaStreamsDlqDispatch = !StringUtils
+
+			ProducerFactory<byte[], byte[]> producerFactory = getProducerFactory(
+					new ExtendedProducerProperties<>(
+							extendedConsumerProperties.getExtension().getDlqProducerProperties()),
+					binderConfigurationProperties);
+			KafkaTemplate<byte[], byte[]> kafkaTemplate = new KafkaTemplate<>(producerFactory);
+
+
+			DeadLetterPublishingRecoverer kafkaStreamsBinderDlqRecoverer = !StringUtils
 					.isEmpty(extendedConsumerProperties.getExtension().getDlqName())
-							? new KafkaStreamsDlqDispatch(
-									extendedConsumerProperties.getExtension()
-											.getDlqName(),
-									binderConfigurationProperties,
-									extendedConsumerProperties.getExtension())
-							: null;
+					? new DeadLetterPublishingRecoverer(kafkaTemplate, (cr, e) -> new TopicPartition(extendedConsumerProperties.getExtension()
+					.getDlqName(), cr.partition()))
```

Comment:
With this logic, the DLT must have the same number of partitions as the original.

We should probably add a new property/properties to allow the user to determine whether the DLQ publish should be done to the same partition, a particular partition, or no partition (kafka chooses).

Reply:
@garyrussell We will address this as a separate issue (maybe after S1P) - Filed [this issue](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/763).